node/src/reactor.rs

```diff
@@ -678,33 +693,40 @@ where
     }
 
     /// Handles dumping queue contents to files in /tmp.
-    async fn dump_queues(&mut self) {
+    async fn dump_queues(&mut self, format: DumpFormat) {
         let timestamp = Timestamp::now();
         self.last_queue_dump = Some(timestamp);
-        let output_fn = format!("/tmp/queue_dump-{}.json", timestamp);
-        let mut serializer = serde_json::Serializer::pretty(match File::create(&output_fn) {
-            Ok(file) => file,
-            Err(error) => {
-                warn!(%error, "could not create output file ({}) for queue snapshot", output_fn);
-                return;
-            }
-        });
-
-        if let Err(error) = self.scheduler.snapshot(&mut serializer).await {
-            warn!(%error, "could not serialize snapshot to {}", output_fn);
-            return;
-        }
 
-        let debug_dump_filename = format!("/tmp/queue_dump_debug-{}.txt", timestamp);
-        let mut file = match File::create(&debug_dump_filename) {
-            Ok(file) => file,
-            Err(error) => {
-                warn!(%error, "could not create debug output file ({}) for queue snapshot", debug_dump_filename);
-                return;
+        match format {
+            DumpFormat::Json => {
+                let output_fn = format!("/tmp/queue_dump-{}.json", timestamp);
+                let mut serializer = serde_json::Serializer::pretty(
+                    match File::create(&output_fn) {
+                        Ok(file) => file,
+                        Err(error) => {
+                            warn!(%error, "could not create output file ({}) for queue snapshot", output_fn);
+                            return;
+                        }
+                    },
+                );
+                if let Err(error) = self.scheduler.snapshot(&mut serializer).await {
+                    warn!(%error, "could not serialize snapshot to {}", output_fn);
+                    return;
+                }
+            }
+            DumpFormat::Debug => {
+                let output_fn = format!("/tmp/queue_dump_debug-{}.txt", timestamp);
+                let mut file = match File::create(&output_fn) {
+                    Ok(file) => file,
+                    Err(error) => {
+                        warn!(%error, "could not create debug output file ({}) for queue snapshot", output_fn);
```

Comment:
Do we need _debug_ here? The file name will already indicate the type of the format.

Reply:
I'd rather be explicit, TBH :/